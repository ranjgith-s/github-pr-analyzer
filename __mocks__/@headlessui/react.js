// __mocks__/@headlessui/react.js
const React = require('react');

function passthrough({ children, ...props }) {
  return React.createElement('div', props, children);
}

module.exports = {
  __esModule: true,
  Card: passthrough,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  HeroUIProvider: passthrough,
};
