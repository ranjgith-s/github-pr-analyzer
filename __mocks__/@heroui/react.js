const React = require('react');

// List of custom props to filter out for DOM elements
const CUSTOM_PROPS = [
  'isCurrent',
  'endContent',
  'onPress',
  'variant',
  'color',
  'size',
  'underline',
  'src',
  'alt',
  'className',
  'children',
];

function filterProps(props) {
  const domProps = { ...props };
  CUSTOM_PROPS.forEach((key) => {
    if (key in domProps && key !== 'className' && key !== 'children') {
      delete domProps[key];
    }
  });
  return domProps;
}

function passthrough({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: props.className }, children);
}

function InputMock(props) {
  return React.createElement('input', filterProps(props));
}

function ButtonMock({ children, onPress, onClick, ...props }) {
  // Map onPress to onClick for DOM, do not pass onPress to DOM
  const handler = onPress || onClick;
  const domProps = { ...filterProps(props), onClick: handler, className: props.className };
  return React.createElement('button', domProps, children);
}

function Spinner() {
  // Only return a simple span for test, not a div with data-testid
  return React.createElement('span', null, 'Loading...');
}

function CardMock({ children, ...props }) {
  return React.createElement('div', { ...filterProps(props), className: (props.className || '') + ' mock-card' }, children);
}

module.exports = {
  __esModule: true,
  Card: CardMock,
  CardHeader: passthrough,
  CardBody: passthrough,
  CardFooter: passthrough,
  Button: ButtonMock,
  Input: InputMock,
  Avatar: passthrough,
  Breadcrumbs: passthrough,
  BreadcrumbItem: passthrough,
  Spinner,
};
