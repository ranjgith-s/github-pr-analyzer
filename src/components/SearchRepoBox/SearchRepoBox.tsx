import React from 'react';
import { Input, Card } from '../ui';

interface RepoOption {
  owner: string;
  repo: string;
  fullName: string;
}

interface Props {
  query: string;
  options: RepoOption[];
  onQueryChange: (value: string) => void;
  onSelect: (option: RepoOption) => void;
}

export default function SearchRepoBox({
  query,
  options,
  onQueryChange,
  onSelect,
}: Props) {
  return (
    <div className="relative w-full max-w-xl">
      <Input
        placeholder="Search GitHub repositories (owner/repo or URL)"
        value={query}
        onChange={(e) => onQueryChange(e.target.value)}
        className="w-full"
        aria-label="Search GitHub repositories"
        clearable
      />
      {options.length > 0 && (
        <Card className="absolute w-full mt-1 z-10 p-0">
          {options.map((opt) => (
            <div
              key={opt.fullName}
              className="flex items-center gap-2 px-3 py-2 cursor-pointer hover:bg-muted/40 border-b last:border-b-0 border-border"
              onClick={() => onSelect(opt)}
            >
              <span className="font-mono text-sm text-foreground">
                {opt.fullName}
              </span>
            </div>
          ))}
        </Card>
      )}
    </div>
  );
}
