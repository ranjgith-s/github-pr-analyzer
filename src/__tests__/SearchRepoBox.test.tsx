import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import SearchRepoBox from '../SearchRepoBox';

describe('SearchRepoBox', () => {
  const options = [
    { owner: 'octocat', repo: 'Hello-World', fullName: 'octocat/Hello-World' },
    { owner: 'test', repo: 'Repo', fullName: 'test/Repo' },
  ];

  it('renders input with placeholder', () => {
    render(
      <SearchRepoBox query="" options={[]} onQueryChange={jest.fn()} onSelect={jest.fn()} />
    );
    expect(screen.getByPlaceholderText(/Search GitHub repositories/i)).toBeInTheDocument();
  });

  it('calls onQueryChange when typing', () => {
    const onQueryChange = jest.fn();
    render(
      <SearchRepoBox query="" options={[]} onQueryChange={onQueryChange} onSelect={jest.fn()} />
    );
    fireEvent.change(screen.getByRole('textbox'), { target: { value: 'octocat' } });
    expect(onQueryChange).toHaveBeenCalledWith('octocat');
  });

  it('renders options and calls onSelect', () => {
    const onSelect = jest.fn();
    render(
      <SearchRepoBox query="" options={options} onQueryChange={jest.fn()} onSelect={onSelect} />
    );
    expect(screen.getByText('octocat/Hello-World')).toBeInTheDocument();
    fireEvent.click(screen.getByText('octocat/Hello-World'));
    expect(onSelect).toHaveBeenCalledWith(options[0]);
  });

  it('matches snapshot', () => {
    const { container } = render(
      <SearchRepoBox query="" options={options} onQueryChange={jest.fn()} onSelect={jest.fn()} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
