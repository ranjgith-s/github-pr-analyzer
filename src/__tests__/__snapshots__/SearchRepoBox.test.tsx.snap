// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRepoBox matches snapshot 1`] = `
<div
  class="relative w-full max-w-xl"
>
  <input
    aria-label="Search GitHub repositories"
    class="w-full"
    placeholder="Search GitHub repositories (owner/repo or URL)"
    value=""
  />
  <div
    class="absolute w-full mt-1 z-10 p-0 mock-card"
  >
    <div
      class="flex items-center gap-2 px-3 py-2 cursor-pointer hover:bg-default-100 border-b last:border-b-0 border-divider"
    >
      <span
        class="font-mono text-sm text-foreground"
      >
        octocat/Hello-World
      </span>
    </div>
    <div
      class="flex items-center gap-2 px-3 py-2 cursor-pointer hover:bg-default-100 border-b last:border-b-0 border-divider"
    >
      <span
        class="font-mono text-sm text-foreground"
      >
        test/Repo
      </span>
    </div>
  </div>
</div>
`;
